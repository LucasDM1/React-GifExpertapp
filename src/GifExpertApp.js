import React, { useState }from 'react';
//import PropTypes from 'prop-types'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({defaultCategories=[]}) => {

    const [categories, setCategories] = useState(defaultCategories);
    //GifExpertApp

    // const handleAdd = () =>{
    //    setCategories([...categories, "HunterxHunter"]); 
    // }
    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr />
            
                {categories.map((category)=>{
                    return(
                        <GifGrid
                        key={category}
                         category={category} />
                    );
                })}
                {/* <button onClick={()=>{handleAdd()}}>Agregar </button> */}
            

        </div>
    );

}