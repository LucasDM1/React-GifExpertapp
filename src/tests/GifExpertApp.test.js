import react from 'react'
import { shallow } from 'enzyme'
import {GifExpertApp} from '../GifExpertApp'


describe('Pruebas de gifExpertApp', () => {

    test('Debe mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />)   
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe mostrar una lista de categorias', () => {
        
        const categories =['Berserk', 'Goku'];

        const wrapper =shallow(<GifExpertApp defaultCategories={categories} />)
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
