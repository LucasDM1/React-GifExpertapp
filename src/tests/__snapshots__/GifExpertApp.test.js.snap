// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de gifExpertApp Debe mostrarse correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</div>
`;
